### DISCUSSION TITLE
nujucks templating library

### DISCUSSION THREAD

**Discussion by kinghat** [Nerwyn/universal-remote-card#40] [Ideas]
if you think this is a problem area that other custom components might come across, have you thought about breaking your work out into a library?

**Comment by Nerwyn** [comment]
I have! I was thinking about splitting it out into a different repo, importing that into my projects, and sharing the repo on the Home Assistant forum in the future. Plus that way I'd hopefully be able to use the community to build out the rest of the Home Assistant templating functions.

**Reply by kinghat** [reply]
im unsure how it all works at this point but im looking to integrate templating into my card. so im sure it would come in handy.

**Comment by Nerwyn** [comment]
I've split it out into it's own npm package, and plan to make minor release versions of my projects to switch them to this.

https://github.com/Nerwyn/ha-nunjucks

https://www.npmjs.com/package/ha-nunjucks

Let me know how it works! I already tested it on my other project and it seems to be fine. I'll make a post about it on the Home Assistant community forums once I officially implement it into my projects.

**Reply by Nerwyn** [reply]
I'm working on adding tests and fixing bugs as I find them now. Once that's done I'm going to make a PR template and contributing guide.

**Reply by Nerwyn** [reply]
ha-nunjucks v1.1.0 is live! The repo is all set up with tests for all functions (except `match-media` which can't really be tested in node) and proper CI/CD so that it'll publish to npmjs on release creation. I've also updated `android-tv-card` and `service-call-tile-feature` to use this package and there seems to be no change in behavior in either of them.

I'll work on in depth contributing instructions tomorrow and then share to the Home Assistant community forums. I'm hoping there's enough interest that other developers will want to code out all of the templating functions that I haven't yet.

TODO:

- [x] Update `android-tv-card` and `service-call-tile-feature` READMEs to link to `ha-nunjucks` for more in depth templating information.
- [x] Create `ha-nunjucks` contributing docs.
- [x] Share to Home Assistant community forums.

**Reply by kinghat** [reply]
thanks for this! i will follow the development over there and ill report anything i find when i implement it my card.
